### PR TITLE
PS-1890 Switch to Azure pipelines

### PR DESCRIPTION
**Pull request by joker806**
https://keboola.atlassian.net/browse/PS-1890

- testing Synapse must be running for tests to pass :(
- on Travis, tests were run on PHP 5.6 & 7.0, now it's on PHP 5.6 & 7.4 (same as input mapping)
- no parallelization was done yet - first we have to slice tests into more independent test suites
- for now, it looks like the speed is roughly the same as on Travis - PHP 5.6 tests took 36 minutes on Azure

### CODE REVIEW

File: tests/Writer/BaseWriterTest.php [hunk 1]
```diff
@@ -28,6 +28,10 @@ abstract class BaseWriterTest extends \PHPUnit_Framework_TestCase
 
     protected function clearBuckets($buckets)
     {
+        if ($this->clientWrapper === null) {
+            return;
+        }
```
Comment (pivnicek) [inline tests/Writer/BaseWriterTest.php:33]: just curious why was this necessary?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
+      context: .
+      args:
+        PHP_VERSION: 7.4
+        XDEBUG_VERSION: 2.9.8
+
+  dev:
+    <<: *tests
+    build:
+      context: .
+      target: dev
```
Comment (pivnicek) [inline docker-compose.yml:35]: learn something new every day, diky pepo! 😄

**Review by pivnicek — APPROVED** [review]
looks good 👍

File: tests/Writer/BaseWriterTest.php (hunk 1 above)
Comment (joker806) [reply tests/Writer/BaseWriterTest.php:33]: Greate question! I forgot to mention it in description - if I ran with `RUN_SYNAPSE_TESTS=0` it was crashing in `tearDown` because of not existing `clientWrapper`. But it's weird because I tried it again now without the workaround and is seem to work properly :thinking: 

I'll run whole testsuite one mor time without the workaround and remove it, if there is no problem. Maybe the root cause was somewhere else.

Comment (pivnicek) [reply tests/Writer/BaseWriterTest.php:33]: ah, maybe those return statements you added make this unnecessary?

File: docker-compose.yml (hunk 2 above)
Comment (joker806) [reply docker-compose.yml:35]: Nothing really new.. I like it as an alternative to having multiple Dockerfiles in a project as it's always a pain to reference between them.

Comment (pivnicek) [reply docker-compose.yml:35]: I mean that the `target` build property is new to me, I like it 👍

File: tests/Writer/BaseWriterTest.php (hunk 1 above)
Comment (joker806) [reply tests/Writer/BaseWriterTest.php]: Tak testy bezi i bez toho workaroundu. Davam cely ten jeden commit pryc, at to zbytecne neprovokuje.
